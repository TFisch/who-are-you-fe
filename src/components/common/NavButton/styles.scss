.nav-button {
  background: none;
  border: none;
  cursor: pointer;
  margin-left: 1em;
  margin-right: 1em;
  padding: 0;
  font-family: 'Josefin Slab', serif;
  font-size: 1rem;
}

.nav-button:hover {
  color: pink;
}
