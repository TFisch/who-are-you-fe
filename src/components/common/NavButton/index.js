import React from 'react';
import './styles.scss';

const NavButton = ({ buttonText, handleClick }) => {
  return (
    <button className="nav-button" onClick={handleClick}>
      <h3>{buttonText}</h3>
    </button>
  );
};

export default NavButton;
