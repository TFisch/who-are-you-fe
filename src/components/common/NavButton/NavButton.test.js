import React from 'react';
import { shallow } from 'enzyme';
import NavButton from './'

describe('NavButton', () => {
  let wrapper;

  beforeEach(() => {
    let mockProps = {
      buttonText: "mock"
    }

    wrapper = shallow(<NavButton props={mockProps}
    />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});