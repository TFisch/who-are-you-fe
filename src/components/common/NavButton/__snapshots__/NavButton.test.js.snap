// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButton should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavButton
    props={
      Object {
        "buttonText": "mock",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h3 />,
      "className": "nav-button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "h3",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3 />,
        "className": "nav-button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
